quarkus/resteasy-qson-common/deployment/src/main/java/io/quarkus/qson/resteasy/deployment/CommonQsonJaxrsBuildStep.java

```diff
@@ -144,8 +145,8 @@ private Method findMethod(MethodInfo method) throws Exception {
             return declaring.getDeclaredMethod(method.name());
         }
         List<Class> params = new LinkedList<>();
-        for (Type type : method.parameters()) {
-            params.add(tcl.loadClass(type.name().toString()));
+        for (MethodParameterInfo info : method.parameters()) {
```

Comment:
You could also just used `method.parameterTypes()`, that would be the equivalent of the previous code.

Reply:
Ah, cool,